pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -244,6 +244,12 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) Results {
 			break
 		}
 
+		// when context has been canceled or deadline exceeded, stop attempting to schedule pods and mark current pod as unschedulable
+		if ctx.Err() != nil {
+			errors[pod] = ctx.Err()
+			break
```

Comment:
I'm curious what you think about this in the context of applying a scheduling timeout as well -- we can choose to either: 1) Mark all pods that have not been scheduled yet as failing due to context id -- this could take some time potentially depending on the number of pods that we have OR 2) Fail the scheduling loop with a context timeout error -- in that case, we'd have to somehow know that we didn't schedule all the pods -- given that we are using errors to communicate this right now, that might not work well

Reply:
In either case, I don't think we can break here because I think we need to mark all pods that weren't scheduled as unschedulable to have a proper interface